### ISSUE TITLE
Handle statutory short cites

### ISSUE THREAD

**Issue by jcushman** [freelawproject/eyecite#77]
Here's an example of statutory short cites:

> Business activities of national banks are controlled by the National Bank Act (NBA or Act), **12 U. S. C. § 1 et seq.**, and regulations promulgated thereunder by the Office of the Comptroller of the Currency (OCC). **See §§24, 93a, 371(a).** As the agency charged by Congress with supervision of the NBA, OCC oversees the operations of national banks and their interactions with customers. See NationsBank of N. C., N. A. v. Variable Annuity Life Ins. Co., 513 U. S. 251, 254, 256 (1995). The agency exercises visitorial powers, including the authority to audit the bank’s books and records, largely to the exclusion of other governmental entities, state or federal. See **§ 484(a); 12 CFR § 7.4000 (2006).**

> The NBA specifically authorizes federally chartered banks to engage in real estate lending. **12 U. S. C. § 371.** It also provides that banks shall have power “[t]o exercise ... all such incidental powers as shall be necessary to carry on the business of banking.” **§24 Seventh**. Among incidental powers, national banks may conduct certain activities through “operating subsidiaries,” discrete entities authorized to engage solely in activities the bank itself could undertake, and subject to the same terms and conditions as those applicable to the bank. See **§ 24a(g)(3)(A); 12 CFR § 5.34(e) (2006)**.

When we see just "§ <foo>" we should potentially fill in the part before § with the previous cite containing §, so "§§ 24" becomes "12 U. S. C §§ 24".

This is interesting because it's not a short cite for clustering purposes ... we want to fill in what is probably the completion of the citation, but not treat them like citations to the same document.

**Comment by munyanjacob** [comment]
I believe this would also apply here:

> 34 Stat.1246'; **U.S.C., § 682, Title 18, and § 345, Title 28**; Acts January 31, 1928,

If the first `U.S.C., § <section>, Title <title>` format was recognized, adding what preceded the § in the full citation to subsequent short citations that begin with § would allow them to be recognized too (assuming that the short cites end in a format the regexes can match).

If the format of the short cite is different, such as something like `U.S.C., § 682, Title 18, ... See § 459.`, then it could still be missed (the new citation `U.S.C., § 459` wouldn't have a title). Perhaps trying to match with the full citation's beginning first, then a variation (e.g. `18 U.S.C.` instead of just `U.S.C.,`) would match these? Or maybe saving the most recent reporter and title for when short cites are found?

**Comment by devlux76** [comment]
Unsure on the coding-end, but on the legal end this could cause problems because, unlike case law, the body of statutory and administrative law are constantly in flux. The U.S.C. isn't the law, it is merely, prima facie evidence of the law.

What I would highly recommend is to keep eyecite focused strictly on case law, and create a separate scanner focused on statutory law and also a third scanner focused on administrative law.  Perhaps they could be chained together into a pipeline, but under the hood these do have distinctly different concerns.  

When a lawyer is looking for case law, they're looking for cases that have substantially similar facts to the case at hand. This is because stare decisis and res judicata mean that if you can find a case that's on-point, binding in your jurisdiction and still "good law" (also called "on all fours"), the judge over your case is required to follow it. Typically, Statutory Law is resorted to directly only when there isn't any case on-point they can cite to. But you have to be super careful with Statutory Law and Administrative Law since they're constantly in flux and judges are as loathe to make precedents as they are to break them. 

First off, most people don't cite the U.S.C., they cite U.S.C.A. (United States Code Annotated) or U.S.C.S. (United States Code Service).

These aren't laws, they're books about the law and sadly they're subject to copyright. Usually, they contain the entire United States Code plus commentary and notes for easy reference. 

Thus the scanner would need to be aware of what precisely it's looking at and more importantly, when.

The actual U.S.C. changes regularly. Every time Congress passes a law it invariably ends up there. But Congress itself doesn't maintain the United States Code. It's rare you see a law that says, "We're going to update Title V to say this now...".  
The [Office of Law Revision Counsel](http://uscode.house.gov/search/help/uschelp.shtml) does most of the work to make it all fit together.

Because it is in flux, we would have to have a concept of when the citation is from. They don't put a version number on it like in software, but we need that or we might never find the correct law because these things do get shifted around and re-ordered from time to time as needed.

For instance the above cite
>34 Stat.1246'; U.S.C., § 682, Title 18, and § 345, Title 28; Acts January 31, 1928,

Is not a citation, it's part of an explanation.

Firstly it's citing [34 Stat. 1246](http://uscode.house.gov/statviewer.htm?volume=34&page=1246) which is a reference to the United States Statutes at Large specifically the [Criminal Appeals Act of 1907](http://uscode.house.gov/popularnames/popularnames.htm#letterC).

Meanwhile, to understand the rest of the cite it should be read as 18 U.S.C. § 682 and 28 U.S.C. § 345  
Which as you can see, [no longer exist](https://www.law.cornell.edu/uscode/text/18/part-I) but which my research indicates were either created or updated by the [Criminal Appeals Act of 1907](http://uscode.house.gov/popularnames/popularnames.htm#letterC).  This appears to be a cite from January 31st, 1928. So if you can find the version of the United States Code in effect back then, it becomes possible to know what it was talking about. Most likely this would be the first edition since the United States Code project was enacted in 1926 and it probably took a bit of time to compile it all.

Compare that to case law, where a mere mention of Miranda Rights, Biven,  Roe v. Wade, Brown v. Board of Education, Twombly & Iqbal all mean actual specific things because they point to decisions by judges about landmark cases and they all have canonical citation forms in the form of <volume> reporter <page>, <pincite> and it becomes clear it's not just a matter of updating a regex unless all you're looking for is a starting point for feature extraction.

Hope that makes sense!

**Comment by devlux76** [comment]
Correct citation for U.S.C. is < title > U.S.C. § < section > according to the 21st Edition of the BlueBook @ R 12.3(b), but yeah there are some crazy citation forms out there.

**Comment by mlissner** [comment]
Yes, I think we're all comfy around here with the fact that cites U.S.C. mean different things on different days of the week (practically), but we'd still like to be able to find them in a block of text. The challenge of linking them to the correct place is real, but not one that eyecite tries to do. That's up to the library that's calling eyecite.

**Comment by devlux76** [comment]
Good point! I deleted my previous comment. 2 years of Law School being so competitive has done a number on my netiquette and I need to remember to slow down and breathe before responding. I can see clearly now that somehow misread what was being said.  Sorry if I've been coming off condescending. That's not my goal. You guys have done a great job and I'm here to contribute code to a project I believe in.

**Comment by mlissner** [comment]
Noooo worries!
